test/decoders/test_video_decoder_ops.py

```diff
@@ -264,63 +393,81 @@ def test_pts_apis_against_index_ref(self, device):
         )
         torch.testing.assert_close(pts_seconds, all_pts_seconds_ref, atol=0, rtol=0)
 
+    @pytest.mark.parametrize("test_ref", (NASA_VIDEO, NASA_AUDIO))
     @pytest.mark.parametrize("device", cpu_and_cuda())
-    def test_get_frames_in_range(self, device):
-        decoder = create_from_file(str(NASA_VIDEO.path))
-        add_video_stream(decoder, device=device)
+    @pytest.mark.parametrize("seek_mode", ("exact", "approximate"))
+    def test_get_frames_in_range(self, test_ref, device, seek_mode):
+        if device == "cuda" and test_ref is NASA_AUDIO:
+            pytest.skip(reason="CUDA decoding not supported for audio")
+        decoder = create_from_file(str(test_ref.path), seek_mode=seek_mode)
+        _add_stream(decoder=decoder, test_ref=test_ref, device=device)
 
         # ensure that the degenerate case of a range of size 1 works
-        ref_frame0 = NASA_VIDEO.get_frame_data_by_range(0, 1)
+        ref_frame0 = test_ref.get_frame_data_by_range(0, 1)
         bulk_frame0, *_ = get_frames_in_range(decoder, start=0, stop=1)
         assert_frames_equal(bulk_frame0, ref_frame0.to(device))
 
-        ref_frame1 = NASA_VIDEO.get_frame_data_by_range(1, 2)
+        ref_frame1 = test_ref.get_frame_data_by_range(1, 2)
         bulk_frame1, *_ = get_frames_in_range(decoder, start=1, stop=2)
         assert_frames_equal(bulk_frame1, ref_frame1.to(device))
 
-        ref_frame389 = NASA_VIDEO.get_frame_data_by_range(389, 390)
-        bulk_frame389, *_ = get_frames_in_range(decoder, start=389, stop=390)
+        last_index = 389 if test_ref is NASA_VIDEO else 203  # TODO ew
```

Comment:
Yeah, I think we should incorporate some concept of number of frames into the `TestContainerFile` util class. That should allow us to clean this and some other hard-coded things up.

Reply:
Yes, we should try to refactor a few things for these tests. I'm leaving that kind of stuff out of scope for the initial PR, so that we can focus on the high-level design / features first.